### DISCUSSION TITLE
MS Edge w3c webdriver standard support in appium-chromium-driver

### DISCUSSION THREAD

**Discussion by iprecali1** [appium/appium#18449] [General]
Is there a plan to support W3C WebDriver standard on MS Edge in the near future with appium-chromium-driver?

Its already working right now if you pass the the msedgedriver as executable but its using MJSONWP

here is code example
```PYTHON
from time import sleep
from appium import webdriver
from appium.options.common import AppiumOptions


options = AppiumOptions()
options.platform_name = "macOS"
options.automation_name = "Chromium"
options.set_capability(
    "appium:executable",
    "/Users/ivan/gitrepos/automation/resources/msedgedriver/111/msedgedriver",
)
options.set_capability("appium:chromedriverPort", 9516)

driver = webdriver.Remote("http://127.0.0.1:4723", options=options)
driver.get("http://localhost:8888/index.html")
sleep(2)
driver.quit()
```

here is appium log
[msedgedriver.log](https://github.com/appium/appium/files/11157036/msedgedriver.log)

**Comment by KazuCocoa** [comment]
Afaik, no, but perhaps it can support parcially with a few works.

- the given executable driver is mandatory (since it does not have autodownload handler)
- Vendor prefix handling as `ms` instead of `goog`
- browserName as `msedge` or `MicrosoftEdge`

**Reply by KazuCocoa** [reply]
Perhaps https://github.com/appium/appium-chromedriver/pull/297 and a few update in the appium-chromium-driver will gives the edge ability. An example is in the PR.

**Reply by KazuCocoa** [reply]
appium-chromium-driver v1.1.0 may work with MSEge.
e.g. https://github.com/appium/appium-chromedriver/pull/297#issue-1656961938
